jenkins/deploy/Jenkinsfile

```diff
@@ -0,0 +1,96 @@
+                    load "${BUILD_PROPERTIES}"
+                    withEnv(["JAVA_HOME=${ tool type: 'jdk', name: "${JDK_VERSION}" }",
+                             "PATH+MAVEN=${tool type: 'maven', name: "${MAVEN_VERSION}"}/bin:${env.JAVA_HOME}/bin"]) {
+                        sh """
+                            mvn clean compile
+                        """
+                    }
+                }
+            }
+        }
+
+        stage("Test") {
+            steps {
+                configFileProvider([configFile(fileId: "iis-build-properties", variable: 'BUILD_PROPERTIES')]) {
+                    load "${BUILD_PROPERTIES}"
+                    withEnv(["JAVA_HOME=${ tool type: 'jdk', name: "$JDK_VERSION" }",
+                             "PATH+MAVEN=${tool type: 'maven', name: "$MAVEN_VERSION"}/bin:${env.JAVA_HOME}/bin"]) {
+                        sh """
+                            mvn test
+                        """
+                    }
+                }
+            }
+        }
+
+        stage("Install") {
+            steps {
+                configFileProvider([configFile(fileId: "iis-build-properties", variable: 'BUILD_PROPERTIES')]) {
+                    load "${BUILD_PROPERTIES}"
+                    withEnv(["JAVA_HOME=${ tool type: 'jdk', name: "${JDK_VERSION}" }",
+                             "PATH+MAVEN=${tool type: 'maven', name: "${MAVEN_VERSION}"}/bin:${env.JAVA_HOME}/bin"]) {
+                        sh """
+                            mvn clean install -DskipTests
+                        """
+                    }
+                }
+            }
+        }
+
+        stage("Upload") {
+            steps {
+                configFileProvider([configFile(fileId: "iis-build-properties", variable: 'BUILD_PROPERTIES')]) {
+                    load "${BUILD_PROPERTIES}"
+                    withEnv(["JAVA_HOME=${ tool type: 'jdk', name: "${JDK_VERSION}" }",
+                             "PATH+MAVEN=${tool type: 'maven', name: "${MAVEN_VERSION}"}/bin:${env.JAVA_HOME}/bin"]) {
+                        sshagent(['iis-deployment-credentials']) {
+                            sh """
+                                cd iis-wf/iis-wf-primary && \
+                                mvn clean package -Poozie-package,deploy \
+                                    -Dworkflow.source.dir=eu/dnetlib/iis/wf/primary/main \
+                                    -DiisConnectionProperties=${WORKSPACE}/connection-properties/iis-cdh5-test-gw.properties
+                            """
+                        }
+                    }
+                }
+            }
+        }
+
+        stage("Deploy") {
+            steps {
+                sshagent(['iis-deployment-credentials']) {
+                    sh """
+                        WORKING_DIR=/tmp/jenkins/`date +%Y-%m-%d` && \
+                        ssh -o StrictHostKeyChecking=no jenkins@iis-cdh5-test-gw.ocean.icm.edu.pl "\
```

Comment:
Strict host checking was again blind copy from some example Jenkinsfile so I've removed it.

Using `iis.hadoop.frontend.user.name` and `iis.hadoop.frontend.host.name` is also a very good idea, so I've implemented this solution. Because connection properties is a java properties file we cannot use `load` Jenkinsfile command to import the file. For properties files there is a dedicated Jenkins command `readProperties` but our Jenkins instance does not provide this command, probably due to missing plugin. So I've had to manually parse the file to a map from parameter key to parameter value. The solution is a bit verbose, though.

Reply:
At the moment it probably does not matter anymore, but the first connection to a new host is interactive with the default `StrictHostKeyChecking`, so not very well suited for automation. That's probably why this is used in examples, although `accept-new` would be the more conservative value than `no`.